engine/src/main/java/org/terasology/persistence/internal/EntitySetDeltaRecorder.java

```diff
@@ -67,7 +67,9 @@ public void onEntityComponentChange(EntityRef entity, Class<? extends Component>
             EntityDelta entityDelta = getOrCreateEntityDeltaFor(entity);
             Component component = entity.getComponent(componentClass);
             Component componentSnapshot = componentLibrary.copy(component);
```

Comment:
this call to `.copy` fails and `componentSnapshot` is `null`... not sure this check should be in here or not, but we face similar a similar issue one step later in `NetworkSystemImpl` ...

Reply:
do we know anything yet about why that fails? or still needs research?